frontend/www/js/omegaup/components/problem/Settings.vue

```diff
@@ -113,20 +112,36 @@ import { T } from '../../omegaup.js';
 import UI from '../../ui.js';
 
 @Component
-export default class ProblemValidator extends Vue {
+export default class ProblemSettings extends Vue {
   @Prop() timeLimit!: number;
   @Prop() extraWallTime!: number;
   @Prop() memoryLimit!: number;
   @Prop() outputLimit!: number;
   @Prop() inputLimit!: number;
   @Prop() overallWallTimeLimit!: number;
   @Prop() validatorTimeLimit!: number;
-  @Prop() languages!: string;
+  @Prop() initialLanguage!: string;
   @Prop() validLanguages!: Array<string>;
-  @Prop() validator!: string;
-  @Prop() validatorsTypes!: Array<string>;
+  @Prop() initialValidator!: string;
+  @Prop() validatorTypes!: Array<string>;
 
   T = T;
   UI = UI;
+
+  get validator(): string {
+    return this.initialValidator;
+  }
+
+  set validator(validator: string) {
+    this.initialValidator = validator;
+  }
+
+  get languages(): string {
+    return this.initialLanguage;
+  }
+
+  set languages(language: string) {
+    this.initialLanguage = language;
+  }
```

Comment:
esto no es necesario. puedes hacer

```suggestion
  validator = this.initialValidator;
  languages = this.initialLanguages;

  @Watch('initialValidator')
  onInitialValidatorChange(newInitial: string): void {
    this.validator = newInitial;
  }

  @Watch('initialLanguages')
  onInitialLanguagesChange(newInitial: string): void {
    this.languages = newInitial;
  }
```

así como en https://github.com/omegaup/omegaup/blob/master/frontend/www/js/omegaup/components/problem/StatementEdit.vue